src/cap32.cpp

```diff
@@ -2151,25 +2121,9 @@ int cap32_main (int argc, char **argv)
               dword cpc_key[2] = {0xff, 0xff};
               bool release = false;
               CPC.InputMapper->CPCkeyFromJoystickAxis(event.jaxis, cpc_key, release);
-			  if (!CPC.paused && cpc_key[0] != 0xff) {
-                if(release) {
-                 keyboard_matrix[static_cast<byte>(cpc_key[0]) >> 4] |= bit_values[static_cast<byte>(cpc_key[0]) & 7]; // key has been released
+              applyKeypress(cpc_key[0], keyboard_matrix, !release);
+              if (!CPC.paused && cpc_key[0] != 0xff && release) {
                  keyboard_matrix[static_cast<byte>(cpc_key[1]) >> 4] |= bit_values[static_cast<byte>(cpc_key[1]) & 7]; // key has been released
```

Comment:
Well, we test CPC.paused and cpc_key != 0xff in the applyKeypress function.
The original code already used cpc_key[0] != 0xff as a condition to release cpc_key[1] - but looking back at the code, you are right, we should have tested that cpc_key[1] is not 0xff anyway.